tests/functional/func_test.py

```diff
@@ -130,10 +129,15 @@ def run_testcase(test_name, config, verbose, test_case):
                 for config_name in test_module.config:
                     testcase_name = "%s.%s[%s]" %(test_module.__name__, obj, config_name)
                     config = test_module.config[config_name]
-                    success = run_testcase(testcase_name, config, verbose, test_case) and success
-            if not success:
-                rc = 1
+                    result = run_testcase(testcase_name, config, verbose, test_case)
```

Comment:
We have test cases with string config as well, please add failed testcases appending here as well.
```
            if type(test_module.config) is str:
                success = run_testcase(test_module.__name__ + '.' + obj, test_module.config, verbose, test_case)	                success = run_testcase(test_module.__name__ + '.' + obj, test_module.config, verbose, test_case)
            elif type(test_module.config) is dict:	            elif type(test_module.config) is dict:
                success = True	                success = True
```